lisa/tests/staging/scheduler/util_tracking.py

```diff
@@ -0,0 +1,338 @@
+        - UP: periodic ramp-up task, to slowly increase `util_avg`
+
+        """
+        failure_reasons = {}
+        metrics = {}
+
+        # We have only two task: the main 'rt-app' task and our 'test_task'
+        test_task = self.trace.analysis.rta.rtapp_tasks[-1]
+
+        ue_df = self.trace.df_events('sched_util_est_task')
+        ue_df = ue_df[ue_df.pid == test_task.pid]
+        ua_df = self.trace.df_events('sched_load_se')
+        ua_df = ua_df[ua_df.pid == test_task.pid]
+
+        failures = []
+        for phase in self.trace.analysis.rta.task_phase_windows(test_task):
+            phase_df = ue_df[phase.start:phase.end]
+            area_enqueued = series_integrate(phase_df.util_est_enqueued)
+            area_ewma = series_integrate(phase_df.util_est_ewma)
+
+            phase_df = ua_df[phase.start:phase.end]
+            area_util = series_integrate(phase_df.util)
+
+            metrics[phase.id] = self.PhaseStats(phase.start, phase.end,
+                                                area_util, area_enqueued, area_ewma)
+
+            phase_name = "phase {}".format(phase.id)
+            if area_enqueued < area_util:
+                failure_reasons[phase_name] = 'Enqueued smaller then Util Average'
+                failures.append(phase.start)
+                continue
+
+            # Running on FastRamp kernels:
+            if self.fast_ramp:
+
+                # STABLE, DOWN and UP:
+                if area_ewma < area_enqueued:
+                    failure_reasons[phase_name] = 'NO_FAST_RAMP: EWMA smaller then Enqueued'
+                    failures.append(phase.start)
+                    continue
+
+            # Running on (legacy) non FastRamp kernels:
+            else:
+
+                # STABLE: ewma ramping up
+                if phase.id == 0 and area_ewma > area_enqueued:
+                    failure_reasons[phase_name] = 'FAST_RAMP(STABLE): EWMA bigger then Enqueued'
+                    failures.append(phase.start)
+                    continue
+
+                # DOWN: ewma ramping down
+                if 0 < phase.id < 5 and area_ewma < area_enqueued:
+                    failure_reasons[phase_name] = 'FAST_RAMP(DOWN): EWMA smaller then Enqueued'
+                    failures.append(phase.start)
+                    continue
+
+                # UP: ewma ramping up
+                if phase.id > 4 and area_ewma > area_enqueued:
+                    failure_reasons[phase_name] = 'FAST_RAMP(UP): EWMA bigger then Enqueued'
+                    failures.append(phase.start)
+                    continue
+
+        bundle = ResultBundle.from_bool(len(failure_reasons)==0)
+        bundle.add_metric("fast ramp", self.fast_ramp)
+        bundle.add_metric("phases stats", metrics)
+        if not failure_reasons:
+            return bundle
+
+        # Plot signals to support debugging analysis
+        self._plot_signals(test_task, 'areas', failures)
```

Comment:
IMO that should be done regardless of the test result - we do plot a few things unconditionally in `load_tracking`.